### DISCUSSION TITLE
Question: Is it possible to create new Resource files already with the T4 custom code generator?

### DISCUSSION THREAD

**Discussion by kk738** [dotnet/ResXResourceManager#535] [Q&A]
I want to change all my resx files to have the T4 generator for designer files creation.
Is it possible to set the custom generator as default, so creating new resource files, already use it?

I it is possible to change the ResXFileCodeGenerator to PublicResXFileCodeGenerator in the project settings by changing the generator, but the property is empty with the T4 generator. How can I achieve this?

https://stackoverflow.com/questions/4274311/visual-studio-resx-file-default-internal-to-public

**Comment by tom-englert** [comment]
You can e.g. use it like that:
https://github.com/dotnet/ResXResourceManager/blob/master/src/Directory.Build.targets

**Comment by kk738** [comment]
Sorry I don't get it, how will this set the T4 generator for new resource files automatically?
Could you explain this in a few sentences?

**Comment by kk738** [comment]
Is there a better way than just copying an existing resource to another folder with its tt file to another location, when creating a new resource?

![image](https://user-images.githubusercontent.com/58267898/220587787-97fa9a4b-a7c5-4f3a-b579-907e1f1f50a0.png)

**Comment by tom-englert** [comment]
If you like to automate this, you could use a build step that copies empty templates if a project has no resources yet.

**Comment by kk738** [comment]
Sorry, I think we are talking past each other. 
I am not talking about an empty project without resources. 

Situation is:
I have a project with resources already using the TT4 files as the generator.

Question is:
How can I set the TT4 file generator automatically for new resources that are created by me?

**Reply by tom-englert** [reply]
Do you create them via Visual Studio UI or by just creating a new, empty resource file?

**Reply by kk738** [reply]
Asking this, I think you you're implying it matters?
I in that case thought about creating it via right click --> new file --> resource file
But also other ways should possibly set the T4 file automatically

I just want to avoid somebody creating new resource files and they having to change it to tt4 files every time, or accidentally forgetting to change the custom tool to the template.

**Reply by tom-englert** [reply]
Actually there is nothing that will do this automatically, however you could achieve this by MSBuild, by properly overriding the generator for .resx files to nothing and linking the .t4 file automatically.
